src/Stratis.Bitcoin.Features.BlockStore/AddressIndexing/AddressIndexCache.cs

```diff
@@ -0,0 +1,199 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using LiteDB;
+
+namespace Stratis.Bitcoin.Features.BlockStore.AddressIndexing
+{
+    /// <summary>The basic principle behind this cache is that it is unnecessary and undesirable to have
+    /// every known address in RAM simultaneously. Typically, only a small subset of the addresses with a
+    /// nonzero balance are expected to be transacting. For example, large staking balances. Therefore
+    /// the number of balance change records that are retained in memory is bounded in order to restrict
+    /// the maximum amount of memory that the indexer consumes. Items in the cache that have been
+    /// modified are flagged as dirty, so that the periodic flushing task only needs to write these
+    /// changes to disk.</summary>
+    public class AddressIndexCache
+    {
+        /// <summary>Default maximum number of items (addresses and balance changes) in the cache.</summary>
+        public const int AddressIndexCacheMaxItemsDefault = 100000;
```

Comment:
Why this number? 
I don't think it will work btw. Not the number but approach. We need to have a limit based on number of changes (each record's weight shouldn't be 1 but a number of addr changes). Otherwise 5 cache items might consume more memory than  100000 items when those 5 are major staking addresses